### ISSUE TITLE
Questions about DGGT training details and compute setup

### ISSUE THREAD

**Issue by yy-oct** [xiaomi-research/dggt#50]
Hi, thank you for releasing the code and answering previous questions.

I noticed that you mentioned in another issue that the complete Waymo dataset could be trained with four H20 GPUs. Could you please share more training details if possible? This would be very helpful for reproducing and fine-tuning the model.

Specifically, I would like to ask:

1. For the experiments reported in the paper, how many GPUs were used, and what GPU type was used?
2. Approximately how long did the full training take?
3. How many epochs or iterations were used for training on Waymo / nuScenes / Argoverse2?
4. In the released `train.py`, `max_epoch` is used as the outer loop over the dataloader. Should this be interpreted as the actual number of epochs?
5. In issue #36, you mentioned that noticeable improvements usually appear after around 200 epochs. Does this refer to the released training setting on Waymo?
6. What were the batch size per GPU, global batch size, sequence length, and image resolution used in your training?
7. Did you use mixed precision, gradient checkpointing, or any other memory-saving strategy?
8. Were the VGGT backbone and camera head frozen throughout training, while only the Gaussian / instance / sky heads were trained?
9. Is the released training command sufficient to reproduce the paper-level results, or are there additional important hyperparameters or training stages?

Thank you very much for your time. Any training details or recommended settings would be greatly appreciated.

**Comment by Ziyi-Xiong1** [comment]
For the training setup, we used 4 GPUs. The full training process took approximately 3–5 days. The detailed training settings are provided as default values in the released code. In addition, the VGGT backbone and camera head were frozen during training, while the trainable heads were optimized.

**Comment by ZaiyiHu** [comment]
> For the training setup, we used 4 GPUs. The full training process took approximately 3–5 days. The detailed training settings are provided as default values in the released code. In addition, the VGGT backbone and camera head were frozen during training, while the trainable heads were optimized.

Hi, thanks for releasing the code for DGGT!

I have a question about the Waymo training split used in the released training script.

In train.py, the dataset is constructed as follows:

https://github.com/xiaomi-research/dggt/blob/main/train.py#L98

dataset = WaymoOpenDataset(
    args.image_dir,
    scene_names=[str(i).zfill(3) for i in range(300,600)],
    sequence_length=args.sequence_length,
    mode=1,
    views=1
)

Since Python range(300,600) is exclusive of 600, this seems to use only scenes 300 to 599, i.e. 300 scenes in total.

However, in the paper, Section 4.1 Experimental protocol / Datasets states that Waymo training uses the full Waymo Open Dataset training split, which contains 798 scenes, and the evaluation split contains 202 scenes.

Could you clarify the intended training setting for reproducing the Waymo results in the paper?

Specifically:

Is the released train.py intended to reproduce the paper's Waymo training results, or is it only a simplified/demo training script?

If we want to train with the full 798 Waymo training scenes, should we change this line to something like:

scene_names=[str(i).zfill(3) for i in range(798)]


Is there a reason why the released script uses only scenes 300–599?

Are there any other training settings that differ from the paper, such as mode=1, views=1, number of input views, target frames, or training iterations?

Thanks a lot!
